force-app/main/default/classes/ERR_Notifier_TEST.cls

```diff
@@ -0,0 +1,335 @@
+        System.assertEquals(true, notifierService.sendEmailAsync, 'sendEmailAsync value should be true');
+    }
+
+    /**************************************************************************************************************************
+     * @description Test method to verify that the getInstance method returns the existing instance of the
+     * ERR_Notifier class when one already exists.
+     ***************************************************************************************************************************/
+    @isTest
+    private static void getInstanceExisting() {
+        Test.startTest();
+        ERR_Notifier notifierServiceInstance1 = ERR_Notifier.getInstance();
+
+        ERR_Notifier notifierServiceInstance2 = ERR_Notifier.getInstance();
+        Test.stopTest();
+
+        System.assertEquals(
+            ERR_Notifier.instance,
+            notifierServiceInstance1,
+            'Instance of service class returned should match static instance variable.'
+        );
+
+        System.assertEquals(
+            notifierServiceInstance1,
+            notifierServiceInstance2,
+            'Subsequent retrievals of service class instance should return existing instance.'
+        );
+
+        System.assertEquals(true, notifierServiceInstance1 != null, 'Instance of service class should not be null.');
+    }
+
+    /**********************************************************************************************
+     * @description Test retrieval of System Administrator Users' Emails
+     * verify: Emails of System Administrator Users are retrieved
+     */
+    @IsTest
+    private static void shouldMatchSystemAdminEmailList() {
+        ERR_Notifier notifierService = ERR_Notifier.getInstance();
+        Test.startTest();
+        Set<String> recipientEmails = new Set<String>(
+            notifierService.getErrorEmailRecipients(ERR_Notifier.notificationOptions.sysAdmins)
+        );
+        Test.stopTest();
+
+        Set<String> expectedEmails = getSystemAdminEmails();
+
+        System.assertEquals(expectedEmails.size(), recipientEmails.size(), 'Number of emails should match');
+        System.assert(expectedEmails.containsAll(recipientEmails), 'Expected and retrieved emails should be the same');
+    }
+
+    /**********************************************************************************************
+     * @description Confirm that the method to retrieve Error log record pending notification
+     * only retrieve records created in the last 48 hours.
+     */
+    @IsTest
+    private static void shouldOnlyRetrieveRecentErrorLogEntries() {
+        ERR_Notifier.MAX_AGE_FOR_ERRORS = Datetime.now().addDays(-3);
+        ERR_Notifier.MAX_RECORDS = 10;
+
+        List<Error__c> errors = new List<Error__c>();
+        // The oldest error will be 80 days old
+        Datetime dt = Datetime.now().addDays(80).addHours(-1);
```

Comment:
Is `addHours(-1)` necessary?

Reply:
I separated this test into 2 different test methods. This was needed to make sure it meets the filter condition which is based on Datetime.now()